### PR TITLE
NH-23: Allow Test User Access to HSS

### PR DESCRIPTION
**Pull request by YuYing-Liang**
## Overview

- Resolves #NH-23
- added the ability for test users to place orders and use HSS

## Unit Tests Created

- added 1 test in hardware/test_api for the group

## Steps to QA

- try to use the hss with the "HSS Test Users" group attached to your user, all actions should work

### CODE REVIEW

File: hackathon_site/event/migrations/0005_hss_test_users.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+def apply_migration(apps, schema_editor):
+    for app_config in apps.get_app_configs():
+        app_config.models_module = True
+        create_permissions(app_config, apps=apps, verbosity=0)
+        app_config.models_module = None
+
+    Group = apps.get_model("auth", "Group")
+    Permission = apps.get_model("auth", "Permission")
+
+    group, _ = Group.objects.get_or_create(name="HSS Test Users")
```
Comment (Leo6Leo) [inline hackathon_site/event/migrations/0005_hss_test_users.py]: After I applied the migration, there is no change in my database. I just manually created the group

File: hackathon_site/dashboard/frontend/src/components/cart/CartSummary/CartSummary.tsx [hunk 2]
```diff
@@ -55,7 +59,7 @@ const CartSummary = () => {
                     cartQuantity === 0 ||
                     cartOrderLoading ||
                     !teamSizeValid ||
-                    isOutsideSignOutPeriod
+                    (!isTestUser && isOutsideSignOutPeriod)
```
Comment (Leo6Leo) [inline hackathon_site/dashboard/frontend/src/components/cart/CartSummary/CartSummary.tsx:62]: tested, working properly.

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx [hunk 3]
```diff
@@ -84,7 +91,7 @@ export const AddToCartForm = ({
             size="large"
             type="submit"
             onClick={handleSubmit}
-            disabled={dropdownNum === 0 || isOutsideSignOutPeriod}
+            disabled={dropdownNum === 0 || (!isTestUser && isOutsideSignOutPeriod)}
```
Comment (Leo6Leo) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx:94]: Tested, working properly.

File: hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx [hunk 4]
```diff
@@ -278,20 +292,23 @@ const MainSection = ({
...
-                        categories.map((category, i) => (
-                            <Chip
-                                label={category}
-                                size="small"
-                                className={styles.categoryItem}
-                                key={i}
-                            />
-                        ))}
-                </div>
+                {categories?.length > 0 && (
```
Comment (Leo6Leo) [inline hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.tsx:295]: Tested with the hardware that has more than 2 categories, working properly.

File: hackathon_site/event/migrations/0005_hss_test_users.py (hunk 1 above)
Comment (Leo6Leo) [inline hackathon_site/event/migrations/0005_hss_test_users.py]: After I applied the migration, there is no change in my database. I just manually created the group

File: hackathon_site/hardware/test_api.py [hunk 5]
```diff
@@ -1061,6 +1061,43 @@ def test_submitting_order_after_end_date(self):
         expected_response = {"non_field_errors": ["Hardware sign out period is over"]}
         self.assertEqual(response.json(), expected_response)
 
+    @override_settings(
+        HARDWARE_SIGN_OUT_START_DATE=datetime.now(settings.TZ_INFO)
+        + relativedelta(days=1)
+    )
+    def test_submitting_order_as_test_user_before_start_date_success(self):
```
Comment (Leo6Leo) [inline hackathon_site/hardware/test_api.py:1067]: do we need a test to test if the user doesn't belong to the test user group?

**Review by Leo6Leo — COMMENTED** [review]
Please see my comment above

File: hackathon_site/event/migrations/0005_hss_test_users.py (hunk 1 above)
Comment (YuYing-Liang) [reply hackathon_site/event/migrations/0005_hss_test_users.py]: I'm not sure what would be the problem 🤔 It works for me and tests seem to require the migration to have been applied.

File: hackathon_site/hardware/test_api.py (hunk 5 above)
Comment (YuYing-Liang) [reply hackathon_site/hardware/test_api.py:1067]: those tests already exists, they are functions `test_submitting_order_after_end_date` `test_submitting_order_before_start_date`

**Review by Mustaballer — APPROVED** [review]

**Pull request merged**